Python/codegen.c

```diff
@@ -758,10 +758,10 @@ _PyCodegen_Body(compiler *c, location loc, asdl_stmt_seq *stmts)
         return SUCCESS;
     }
     Py_ssize_t first_instr = 0;
-    if (!IS_INTERACTIVE(c)) {
-        PyObject *docstring = _PyAST_GetDocString(stmts);
-        if (docstring) {
-            first_instr = 1;
+    PyObject *docstring = _PyAST_GetDocString(stmts);
+    if (docstring) {
+        first_instr = 1;
+        if (!IS_INTERACTIVE(c)) {
```

Comment:
Why does this depend on interactive mode at all? I see that class docstrings are removed on current main in the REPL, but I'm not sure why that would happen, and it doesn't match the behavior on 3.12.

Reply:
It seems to have been added here: https://github.com/python/cpython/commit/9b8611eeea172cd4aa626ccd1ca333dc4093cd8c#diff-ebc983d9f91e5bcf73500e377ac65e85863c4f77fd5b6b6caf4fcdf7c0f0b057R1506

Reply:
Oops. I don't know why I did that, I don't think it makes sense. I can take over this fix if you like since it's my bug.

Reply:
It's not your bug. Once you commit it it's our bug.

Reply:
So I just remove the `if (!IS_INTERACTIVE(c)` ?

Reply:
I think so, the condition should just be removed. My best guess is I was messing around with some interaction between annotations and interactive mode, and ended up with this change. The code block deals purely with docstrings and not annotations though, so I should never have made change.

And yes, it's not just my bug now because it's been merged, but I still feel a sense of responsibility for what I committed :)

Reply:
It a nice bug, one that nobody will ever care about.